akka-docs/src/main/paradox/includes/actor-context.md

```diff
@@ -0,0 +1,15 @@
+
+<!--- #actor-context-typed-access --->
+## Accessing the ActorContext
+
+If the behavior needs to use the `ActorContext`, for example to spawn child actors, or use
+@scala[`context.self`]@java[`context.getSelf()`], it can be obtained by wrapping construction with `Behaviors.setup`:
```

Comment:
Hmm, the Java example doesn't show `Behaviors.setup` at all - can we make that consistent?

Reply:
Maybe start the section with an introduction to what ActorContext is used for like a bullet list:

* spawning child actors
* watching other actors
* logging
* creating message adapters
* request-response interaction (ask) with another actor
* access to the self ActorRef

Reply:
@raboof the Java sample shows: `return Behaviors.setup(ctx -> new MyPersistentBehavior(persistenceId, ctx))` below, is this ok?
```java
// #actor-context
    public class MyPersistentBehavior extends EventSourcedBehavior<Command, Event, State> {

      public static Behavior<Command> behavior(PersistenceId persistenceId) {
        return Behaviors.setup(ctx -> new MyPersistentBehavior(persistenceId, ctx));
      }

      // this makes the context available to the command handler etc.
      private final ActorContext<Command> ctx;

      // optionally if you need `ActorContext.getSelf()`
      private final ActorRef<Command> self;

      public MyPersistentBehavior(PersistenceId persistenceId, ActorContext<Command> ctx) {
        super(persistenceId);
        this.ctx = ctx;
        this.self = ctx.getSelf();
      }

      // #actor-context
```